pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/AbstractApexNode.java

```diff
@@ -4,27 +4,56 @@
 
 package net.sourceforge.pmd.lang.apex.ast;
 
+import java.util.List;
+
 import net.sourceforge.pmd.annotation.InternalApi;
 import net.sourceforge.pmd.lang.ast.SourceCodePositioner;
 
-import apex.jorje.data.Location;
-import apex.jorje.data.Locations;
-import apex.jorje.semantic.ast.AstNode;
-import apex.jorje.semantic.exception.UnexpectedCodePathException;
-import apex.jorje.semantic.symbol.type.TypeInfo;
+import com.google.summit.ast.Node;
 
 /**
  * @deprecated Use {@link ApexNode}
  */
 @Deprecated
 @InternalApi
-public abstract class AbstractApexNode<T extends AstNode> extends AbstractApexNodeBase implements ApexNode<T> {
+public abstract class AbstractApexNode extends AbstractApexNodeBase implements ApexNode<Void> {
+
+    /**
+     * {@link AbstractApexNode} wrapper around a single {@link Node}.
+     *
+     * @deprecated Use {@link ApexNode}
+     */
+    @Deprecated
+    @InternalApi
+    public static abstract class Single<T extends Node> extends AbstractApexNode {
```

Comment:
This is a point worth calling out. Since even the (internal) mapping between PMD and the Summit AST isn't 1:1, we use subclass for the different relations between the two. For example, most nodes continue to wrap a `Single` AST node, but there are a few that wrap `Many`.